audio-echo/app/src/main/cpp/audio_player.cpp

```diff
@@ -179,14 +179,18 @@ AudioPlayer::~AudioPlayer() {
   while (devShadowQueue_->front(&buf)) {
     buf->size_ = 0;
     devShadowQueue_->pop();
-    freeQueue_->push(buf);
+    if(buf != &silentBuf_) {
+      freeQueue_->push(buf);
+    }
   }
   delete devShadowQueue_;
 
   while (playQueue_->front(&buf)) {
     buf->size_ = 0;
     playQueue_->pop();
-    freeQueue_->push(buf);
+    if(buf != &silentBuf_) {
```

Comment:
silent buffer probably never get into the "player queue": only recorder fill the player queue for the recorded audio. 

This one may not be necessary, may you help double check it?  thanks